src/core/ddsi/include/dds/ddsi/ddsi_cfgelems.h

```diff
@@ -0,0 +1,1994 @@
+    DESCRIPTION(
+      "<p>This element controls whether Cyclone DDS defaults to deaf mode or to "
+      "normal mode. This controls both the initial behaviour and what "
+      "behaviour it auto-reverts to.</p>")),
+  BOOL(DEPRECATED("Mute"), NULL, 1, "false",
+    MEMBER(initial_mute),
+    FUNCTIONS(0, uf_deaf_mute, 0, pf_boolean),
+    DESCRIPTION(
+      "<p>This element controls whether Cyclone DDS defaults to mute mode or to "
+      "normal mode. This controls both the initial behaviour and what "
+      "behaviour it auto-reverts to.</p>")),
+  END_MARKER
+};
+
+static struct cfgelem rediscovery_blacklist_duration_attrs[] = {
+  BOOL("enforce", NULL, 1, "false",
+    MEMBER(prune_deleted_ppant.enforce_delay),
+    FUNCTIONS(0, uf_boolean, 0, pf_boolean),
+    DESCRIPTION(
+      "<p>This attribute controls whether the configured time during which "
+      "recently deleted participants will not be rediscovered (i.e., \"black "
+      "listed\") is enforced and following complete removal of the "
+      "participant in Cyclone DDS, or whether it can be rediscovered earlier "
+      "provided all traces of that participant have been removed already.</p>"
+    )),
+  END_MARKER
+};
+
+static struct cfgelem heartbeat_interval_attrs[] = {
+  STRING("min", NULL, 1, "5 ms",
+    MEMBER(const_hb_intv_min),
+    FUNCTIONS(0, uf_duration_inf, 0, pf_duration),
+    DESCRIPTION(
+      "<p>This attribute sets the minimum interval that must have passed "
+      "since the most recent heartbeat from a writer, before another "
+      "asynchronous (not directly related to writing) will be sent.</p>"),
+    UNIT("duration_inf")),
+  STRING("minsched", NULL, 1, "20 ms",
+    MEMBER(const_hb_intv_sched_min),
+    FUNCTIONS(0, uf_duration_inf, 0, pf_duration),
+    DESCRIPTION(
+      "<p>This attribute sets the minimum interval for periodic heartbeats. "
+      "Other events may still cause heartbeats to go out.</p>"),
+    UNIT("duration_inf")),
+  STRING("max", NULL, 1, "8 s",
+    MEMBER(const_hb_intv_sched_max),
+    FUNCTIONS(0, uf_duration_inf, 0, pf_duration),
+    DESCRIPTION(
+      "<p>This attribute sets the maximum interval for periodic heartbeats.</p>"),
+    UNIT("duration_inf")),
+  END_MARKER
+};
+
+static struct cfgelem liveliness_monitoring_attrs[] = {
+  BOOL("StackTraces", NULL, 1, "true",
+    MEMBER(noprogress_log_stacktraces),
+    FUNCTIONS(0, uf_boolean, 0, pf_boolean),
+    DESCRIPTION(
+      "<p>This element controls whether or not to write stack traces to the "
+      "DDSI2 trace when a thread fails to make progress (on select platforms "
+      "only).</p>")),
+  STRING("Interval", NULL, 1, "1s",
+    MEMBER(liveliness_monitoring_interval),
+    FUNCTIONS(0, uf_duration_100ms_1hr, 0, pf_duration),
+    DESCRIPTION(
+      "<p>This element controls the interval at which to check whether "
+      "threads have been making progress.</p>"),
+    UNIT("duration"),
+    RANGE("100ms;1hr")),
```

Comment:
Was this present in q_config.c?

Reply:
I thought I'd save myself the hassle of looking up the ranges in a separate table as done in `makernc.pl` and specify it as a field instead (`ddsconf` only, not for `ddsi`). The function `uf_duration_100ms_1hr` gets translated by the script to `100ms;1hr` (as the name correctly suggests :smile:).